lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -419,14 +420,26 @@ private function generateMagicGet(ClassMetadata $class)
         $hasParentGet         = false;
         $returnReference      = '';
         $inheritDoc           = '';
+        $parametersBuffer     = '$name';
+        $firstParameterName   = '$name';
+        $returnTypeHint       = null;
 
         if ($reflectionClass->hasMethod('__get')) {
-            $hasParentGet = true;
-            $inheritDoc   = '{@inheritDoc}';
+            $hasParentGet     = true;
+            $inheritDoc       = '{@inheritDoc}';
+            $methodReflection = $reflectionClass->getMethod('__get');
 
-            if ($reflectionClass->getMethod('__get')->returnsReference()) {
+            if ($methodReflection->returnsReference()) {
                 $returnReference = '& ';
             }
+
+            $methodParameters = $methodReflection->getParameters();
+            if (isset($methodParameters[0])) {
+                $firstParameterName = '$' . $methodParameters[0]->getName();
+            }
+
+            $parametersBuffer = $this->getMethodParametersWithTypes($methodReflection);
```

Comment:
You should be able to use `buildParametersString` for this.

Reply:
@alcaeus Fixed, I am very sorry.